### PR TITLE
Created a Retip component

### PR DESCRIPTION
**Pull request by stanislav-slavov**

### CODE REVIEW

File: src/components/tipRecords/TipRecordComponent.vue [hunk 1]
```diff
@@ -51,23 +52,29 @@
...
     computed: {
       ...mapGetters(['current'])
     },
     methods: {
-      async retip(id) {
-        const amount = util.aeToAtoms(prompt("Tip Amount in AE?"));
+      toggleRetip(flag){
+        this.showRetipWindow = flag
+      },
+      async retip(id, amount) {
```
Comment (thepiwo) [inline src/components/tipRecords/TipRecordComponent.vue]: why keep implementation of retip here and then pass it into the retip component?

File: src/components/RetipComponent.vue [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      ...mapGetters(['current']),
+      toggleForm(){
+        this.resetForm();
+        return this.show;
+      }
+    },
+    methods:{
+      submitRetip(){
+        this.retip(this.tipid, this.value)
+        this.$emit('toggleRetip', false)
```
Comment (thepiwo) [inline src/components/RetipComponent.vue]: maybe this can be better done using props?

File: src/components/RetipComponent.vue [hunk 3]
```diff
@@ -0,0 +1,111 @@
+<template>
+  <div v-if="toggleForm" class="clearfix retip__container">
+    <div class="input-group mr-1 float-left">
+      <input type="number" step="0.000001" v-model="value" @input="onValueChange" class="form-control" aria-label="Default">
+      <div class="input-group-append">
+        <span class="input-group-text append__ae"> <span class="ae">AE</span>&nbsp;(~ {{current.currency.toUpperCase()}} {{ fiatValue }} )</span>
```
Comment (mradkov) [inline src/components/RetipComponent.vue]: You can use `fiat-value` component here for calculating and displaying the fiat value.

File: src/components/RetipComponent.vue [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    components: {
+      'fiat-value': FiatValue
+    },
+    computed: {
+      ...mapGetters(['current']),
+    },
+    methods:{
+      toggleRetip(flag){
+        this.show = flag;
+        if(flag){
```
Comment (thepiwo) [inline src/components/RetipComponent.vue]: lets use a semantic name for "flag"

File: src/components/RetipComponent.vue [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        this.show = false;
+      },
+      resetForm(){
+        this.value = 0;
+        this.fiatValue = 0.00;
+      },
+      onValueChange(data){
+        this.getFiatVal();
+      },
+      async getFiatVal(){
```
Comment (thepiwo) [inline src/components/RetipComponent.vue]: whats the need for all the fiat val logic in here? there is a fiat val component.

File: src/components/RetipComponent.vue [hunk 6]
```diff
@@ -0,0 +1,146 @@
+<template>
+<div class="position-relative wrapper">
+  <img @click="toggleRetip(!show)" class="retip__icon" src="../assets/heart.svg">
+  <div class="clearfix retip__container" v-if="show">
+    <div class="input-group mr-1 float-left">
+      <input type="number" step="0.000001" v-model="value" @input="onValueChange" class="form-control" aria-label="Default">
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply src/components/RetipComponent.vue]: onValueChange is probably not needed if you use computed?

File: src/components/RetipComponent.vue [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+      <div class="input-group-append">
+        <span class="input-group-text append__ae"> <span class="ae">AE</span>&nbsp;<fiat-value :amount="value"></fiat-value></span>
```
Comment (thepiwo) [inline src/components/RetipComponent.vue]: lets use the fiat value component

File: src/components/tipRecords/TipRecordComponent.vue [hunk 8]
```diff
@@ -17,11 +17,11 @@
...
         <a class="tip__url mb-2 text-ellipsis pr-2" :title="tip.url" :href='tip.url' target="_blank">{{tip.url}}</a>
       </div>
-      <div class="tip__footer text-ellipsis">
+      <div class="tip__footer">
         <div class="row">
           <div class="col-lg-9 col-md-12">
-            <span class="tip__amount" @click="retip(tip.id)" title="Send AE to this post">
-              <img src="../../assets/heart.svg"> {{ tip.total_amount }} <span>AE</span>
+            <span class="tip__amount position-relative" >
+              <span class="tip__amount__btn" title="Send AE to this post"><retip-component :tipid="tip.id"></retip-component></span> {{ tip.total_amount }} <span class="ae">AE</span>
```
Comment (davidyuk) [inline src/components/tipRecords/TipRecordComponent.vue]: This line seems to be too long, proposing to split it:
```suggestion
              <span class="tip__amount__btn" title="Send AE to this post">
                <retip-component :tipid="tip.id" />
              </span>
              {{ tip.total_amount }} <span class="ae">AE</span>
```

File: src/components/tipRecords/TipRecordComponent.vue [hunk 9]
```diff
@@ -47,29 +47,28 @@
...
       ...mapGetters(['current'])
     },
     methods: {
-      async retip(id) {
-        const amount = util.aeToAtoms(prompt("Tip Amount in AE?"));
-        await aeternity.contract.methods.retip(id, {amount: amount}).catch(console.error);
-        EventBus.$emit('reloadData');
+      toggleRetip(flag){
+        this.showRetipWindow = flag
       },
```
[the remark this answers is not in the document]
Comment (stanislav-slavov) [reply src/components/tipRecords/TipRecordComponent.vue]: It is not used, thanks for pointing it out.

**Pull request merged**